### PR TITLE
[Under testing] [Tizen.native] Object detection overlay example

### PR DESCRIPTION
**Pull request by kparichay**
Final compilable code
App Logic :
- Set the camera on but don't display it
- Capture the frame from the camera at its preview
- Process the capture frame through the pipeline and display it
- The model file is not added in the commit -
download from https://github.com/nnsuite/testcases/blob/master/DeepLearningModels/tensorflow-lite/ssd_mobilenet_v2_coco/ssd_mobilenet_v2_coco.tflite

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

**Comment by myungjoo** [comment]
@anyj0527 Please test this app in 1. Tizen SDK Emulator and 2. Tizen mobile device.

**Comment by myungjoo** [comment]
@kparichay Please add a README.md.

File: Tizen.native/ObjectDetection/inc/main.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/**
+ * @file main.h
```
Comment (leemgs) [inline Tizen.native/ObjectDetection/inc/main.h]: Looks good.

File: Tizen.native/ObjectDetection/src/data.c [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+  ml_tensors_data_create (info, &data);
+  ml_tensors_data_set_tensor_data (data, 0, frame->data.triple_plane.y, data_size);
+  ml_pipeline_src_input_data (srchandle, data, ML_PIPELINE_BUF_POLICY_AUTO_FREE);
+}
+
+/**
+ * @brief Updates the display object based on the output of the pipeline
+ */
+void
+update_gui(void * data, void * buf, unsigned int size)
```
Comment (leemgs) [inline Tizen.native/ObjectDetection/src/data.c]: Neat.

**Comment by myungjoo** [comment]
To reviewers: do not merge this until we test the app in real device and emulator.

**Comment by anyj0527** [comment]
Well it was built w/o error on tizen-studio, but does not run on my TM1 device (tizen 5.5)...

**Comment by kparichay** [comment]
> Well it was built w/o error on tizen-studio, but does not run on my TM1 device (tizen 5.5)...

@anyj0527 Can you please share the logs? Also please ensure that you have downloaded the model file from [here](https://github.com/nnsuite/testcases/tree/master/DeepLearningModels/tensorflow-lite/ssd_mobilenet_v2_coco) (it's not included in this commit because of the size limitation).

**Comment by anyj0527** [comment]
@kparichay In the tizen-manifest.xml, the api-version set to 2.4, and the Tizen feature machine_learning is not declared. And the `detect.h` seems not used. Is it intended?
I set the version to 5.5 and declared the features machine_learning and machine_learning.inference but the app still not works.

All I got is this crash info:
Callstack Information (PID:24220)
Call Stack Count: 5
 0: _Unwind_VRS_Pop + 0x185 (0xb65ddf56) [/usr/lib/libgcc_s.so.1] + 0x16f56
 1: (0xb65de377) [/usr/lib/libgcc_s.so.1] + 0x17377
 2: (0xb65ddb25) [/usr/lib/libgcc_s.so.1] + 0x16b25
 3: (0xb65ddaa3) [/usr/lib/libgcc_s.so.1] + 0x16aa3
 4: _Unwind_Backtrace + 0x1a (0xb65de233) [/usr/lib/libgcc_s.so.1] + 0x17233
End of Call Stack

I couldn't find logs tagged with "objectdetection".

File: Tizen.native/ObjectDetection/.tproject [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<tproject xmlns="http://www.tizen.org/tproject">
```
Comment (leemgs) [inline Tizen.native/ObjectDetection/.tproject]: @kparichay , What is a meaning of **http://www.tizen.org/tproject** in the 'tproject' tag?

File: Tizen.native/ObjectDetection/.tproject [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    <platforms>
+        <platform>
+            <name>mobile-5.5</name>
```
Comment (leemgs) [inline Tizen.native/ObjectDetection/.tproject]: How about we  work this task based  on mobile-6.0 (w/o mobile-5.5)?

File: Tizen.native/ObjectDetection/res/coco_labels_list.txt [hunk 5]
```diff
@@ -0,0 +1,91 @@
+???
```
Comment (leemgs) [inline Tizen.native/ObjectDetection/res/coco_labels_list.txt]: It's weird. I saw the lots of "???" in this file. Is the **coco_labels_list.txt** file valid?

File: Tizen.native/ObjectDetection/src/data.c [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+   * We only display the modified output image.
+   */
+  error_code = camera_set_display(cam_data.g_camera, CAMERA_DISPLAY_TYPE_NONE,
+      GET_DISPLAY(cam_data.cam_display));
+  if (CAMERA_ERROR_NONE != error_code) {
+    DLOG_PRINT_ERROR("camera_set_display", error_code);
+    return;
+  }
+
+  error_code = camera_set_preview_resolution(cam_data.g_camera, 640,
```
Comment (leemgs) [inline Tizen.native/ObjectDetection/src/data.c]: @helloahn  PTAL, you need to review functions implemented  for  enabling the camera device in ths **data.c** file.

File: Tizen.native/ObjectDetection/inc/detect.h [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+ * @brief Data structure for SSD model info.
+ */
+//typedef struct
+//{
+//  gchar *model_path;       /**< tflite model file path */
+//  gchar *label_path;       /**< label file path */
+//  gchar *box_prior_path;   /**< box prior file path */
+//  gfloat box_priors[BOX_SIZE][DETECTION_MAX];   /**< box prior */
+//  GList *labels;           /**< list of loaded labels */
+//} ssd_model_info_s;
```
Comment (dongju-chae) [inline Tizen.native/ObjectDetection/inc/detect.h]: Leaving them as comments is intended, not removing?

File: Tizen.native/ObjectDetection/.tproject (hunk 4 above)
Comment (kparichay) [reply Tizen.native/ObjectDetection/.tproject]: tizen studio does not offer mobile-6.0 as of now.

File: Tizen.native/ObjectDetection/res/coco_labels_list.txt (hunk 5 above)
Comment (kparichay) [reply Tizen.native/ObjectDetection/res/coco_labels_list.txt]: The file is directly taken from [testcases](https://github.com/nnsuite/testcases/blob/master/DeepLearningModels/tensorflow-lite/ssd_mobilenet_v2_coco) repo of nnsuite

File: Tizen.native/ObjectDetection/inc/detect.h (hunk 7 above)
Comment (kparichay) [reply Tizen.native/ObjectDetection/inc/detect.h]: Removed as this file was wrongly added.

File: Tizen.native/ObjectDetection/.tproject (hunk 3 above)
Comment (kparichay) [reply Tizen.native/ObjectDetection/.tproject]: This file was automatically generated by tizen studio. I am not sure either about the exact meaning of this tag and its value.

File: Tizen.native/ObjectDetection/res/coco_labels_list.txt (hunk 5 above)
Comment (leemgs) [reply Tizen.native/ObjectDetection/res/coco_labels_list.txt]: @helloahn  PTAL,  Could you tell us on the  "**???**" meaning that you uploaded the [testcases](https://github.com/nnsuite/testcases/blob/master/DeepLearningModels/tensorflow-lite/ssd_mobilenet_v2_coco) repo of nnsuite.?

**Comment by myungjoo** [comment]
@anyj0527  has fixes for this. He will send PR after this.

**Pull request merged**